frontend/src/views/MapView.vue

```diff
@@ -3,63 +3,66 @@
     <v-progress-circular indeterminate :size="128"></v-progress-circular>
   </v-overlay>
 
-  <v-container v-if="!mdAndDown" fluid>
-    <v-row fill-height style="height: calc(100vh - 165px)">
+  <v-container fluid>
+    <div id="div-plot-button" class="desktop-plot-buttons-container">
+      <v-card
+        location="left"
+        variant="flat"
+        :style="{
+          backgroundColor: 'transparent'
+        }"
+        max-width="500"
+        max-height="145"
+      >
+        <v-btn
+          style="margin-right: 10px"
+          @click="toggleMetadata"
+          :color="showMetadata ? 'blue' : 'white'"
+        >
+          Metadata
+        </v-btn>
+        <v-btn
+          id="btn-show-historical"
+          style="margin-right: 10px"
+          @click="toggle('historical')"
+          :color="showHistorical ? 'blue' : 'white'"
+        >
+          Historical
+        </v-btn>
+        <v-btn
+          style="margin-right: 10px"
+          @click="toggle('forecast')"
+          :color="showForecast ? 'blue' : 'white'"
+        >
+          Forecast
+        </v-btn>
+      </v-card>
+    </div>
+
+    <v-row
+      fill-height
+      :class="{ 'desktop-map-container': !mdAndDown, 'mobile-map-container': mdAndDown }"
+    >
       <v-col style="padding: 0px; margin: 0px">
         <TheLeafletMap />
       </v-col>
     </v-row>
-  </v-container>
-  <v-container v-else>
-    <v-row style="height: 40vh">
-      <TheLeafletMap />
-    </v-row>
-  </v-container>
 
-  <v-card location="left" :style="getCardStyle()" max-width="500" max-height="145">
-    <v-btn
-      style="margin-right: 10px"
-      @click="toggleMetadata"
-      :color="showMetadata ? 'blue' : 'white'"
-    >
-      Metadata
-    </v-btn>
-    <v-btn
-      style="margin-right: 10px"
-      @click="toggleHistorical"
-      :color="showHistorical ? 'blue' : 'white'"
-    >
-      Historical
-    </v-btn>
-    <v-btn
-      style="margin-right: 10px"
-      @click="toggleForecast"
-      :color="showForecast ? 'blue' : 'white'"
-    >
-      Forecast
-    </v-btn>
-  </v-card>
-
-  <v-card
-    location="left"
-    max-width="500"
-    max-height="800"
-    :style="{
-      transform: 'translate(1vw, -25vh)',
-      position: 'absolute',
-      backgroundColor: 'transparent',
-      'z-index': '9999'
-    }"
-    v-show="showHistorical"
-  >
-    <HistoricalPlot
-      ref="historicalPlotRef"
-      :style="{ width: '500px', height: '300px', padding: '0px 10px' }"
-    />
-  </v-card>
+    <div :class="{ 'mobile-plot-container': mdAndDown, 'desktop-plot-container': !mdAndDown }">
+      <HistoricalPlot
+        v-show="showHistorical"
+        ref="historicalPlotRef"
+        :style="{ width: '500px', height: '300px', padding: '0px 10px', margin: '10px 0px' }"
+      />
+
+      <ForecastPlot
+        v-show="showForecast"
+        ref="forecastPlotRef"
+        :style="{ width: '500px', height: '300px', padding: '0px 10px', margin: '10px 0px' }"
+      />
```

Comment:
My only constructive comment is that there is a lot of duplication between these 2 components.
Not sure if it is worth trying to use the same component for each of them and just pass in different options and data as props...

Reply:
That's a good idea. Given the short deadline, perhaps we make this change in a separate PR though.